docs/DeveloperGuide.md

```diff
@@ -1855,42 +1855,696 @@ testers are expected to do more _exploratory_ testing.
+      Expected: Similar to previous.
+
+   6. Test Case: `login --user Gabriel`{.swift}.</br>
+      Expected: The user does not get logged-in. Error indicating invalid command format is
+      shown in the feedback message.
+
+
+### Update Account Details
+
+1. Updating account details
+
+   1. Prerequisites: Logged-in into the application with the `login`{.swift} command. 
+
+   2. Test Case: `update --user Gabriel123`{.swift}.</br>
+      Expected: The username of the currently logged-in user is updated to `Gabriel123`{.swift}.
+
+   3. Test Case: `update --password GabrielIsCool --confirmPass GabrielIsCool`{.swift}.</br>
+      Expected: The password of the currently logged-in user is updated to `GabrielIsCool`{.swift}.
+
+   4. Test Case: `update --secretQn Favourite Pet --answer Bobo`{.swift}.</br>
+      Expected: The secret question of the currently logged-in user is updated to `GabrielIsCool`{.swift} and the
+      answer is updated to `Bobo`{.swift}.
+
+   5. Test Case: `update`{.swift}.</br>
+      Expected: No user details are updated. Error indicating that at least one field must be specified is
+      shown in the feedback message.
+
+   6. Test Case: `update --user G_briel123`{.swift}.</br>
+      Expected: No user details are updated. Error indicating username constraints is
+      shown in the feedback message.
+
+   7. Test Case: `update --password Gabriel_IsCool --confirmPass Gabriel_IsCool`{.swift}.</br>
+      Expected: No user details are updated. Error indicating password constraints is
+      shown in the feedback message.
+
+   8. Test Case: `update --password GabrielIsCool --confirmPass GabrielIsNotCool`{.swift}.</br>
+      Expected: No user details are updated. Error indicating password and confirm password do not match is
+      shown in the feedback message.
+
+   9. Test Case: `update --password GabrielIsCool`{.swift}.</br>
+      Expected: No user details are updated. Error indicating that password and confirm password must be both present 
+      or both absent is shown in the feedback message.
+
+   10. Test Case: `update --secretQn Favourite Pet`{.swift}.</br>
+       Expected: No user details are updated. Error indicating that secret question and answer must be both present
+       or both absent is shown in the feedback message.
+
+### Logout
+
+1. Logging out of the application
+
+   1. Prerequisites: Logged-in into the application with the `login`{.swift} command.
+
+   2. Test Case: `logout`{.swift}.</br>
+      Expected: The currently logged-in user is logged out of the application, currently displayed Customers/Deliveries
+      are hidden.
+
+   3. Test Case: `logout extra`{.swift} or other extra arguments.</br>
+      Expected: Similar to previous.
+
+### Recover Account
+
+1. Recovering user account
+
+   1. Prerequisites: Logged-in into the application with the `login`{.swift} command, and the answer of the currently
+   logged-in user's secret question is "Koko".
```

Comment:
Would it be better to leave out the logged-in state, as recover account should ideally be performed when logged out?